### PR TITLE
chart-releaser: init at v1.6.1

### PR DESCRIPTION
**Pull request by gnuion**
## Description of changes
Helps Turn GitHub Repositories into Helm Chart Repositories
- [ ] add helm-releaser

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  version = "1.6.1";
+
+  # 0102fa30dd78df0d0f8093c4607e6080972ff82d
+
+  src = fetchFromGitHub {
+    owner = "helm";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-8+O9JErEB1Z/zlrWm975v5Qf0YG0lbPcjY5LlDKw8U4=";
+
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:22]:
```suggestion
```

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+, kubernetes-helm
+, lib
+, makeWrapper
+}:
+
+buildGoModule rec {
+  pname = "chart-releaser";
+  version = "1.6.1";
+
+  # 0102fa30dd78df0d0f8093c4607e6080972ff82d
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:15]: This some kind of checksum? Do we need this comment here, is it useful?

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  # Don't run tests.
+  doCheck = false;
+  doInstallCheck = false;
+
+  vendorHash = "sha256-S/V1kTgD/cXaJNYpPPNjM9ya2zv6Bsy/YBn7I/1EjwI=";
+
+  postPatch = ''
+    substituteInPlace pkg/config/config.go \
+      --replace "\"/etc/ct\"," "\"$out/etc/ct\","
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:33]: > substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'pkg/config/config.go')
substituteStream(): WARNING: pattern \"/etc/ct\"\, doesn't match anything in file 'pkg/config/config.go'

```suggestion
    substituteInPlace pkg/config/config.go \
      --replace-fail "/etc/cr" "$out/etc/cr"
```

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    "-X github.com/helm/chart-releaser/v3/ct/cmd.BuildDate=19700101-00:00:00"
+  ];
+
+  nativeBuildInputs = [ installShellFiles makeWrapper ];
+
+  postInstall = ''
+    installShellCompletion --cmd cr \
+      --bash <($out/bin/ct completion bash) \
+      --zsh <($out/bin/ct completion zsh) \
+      --fish <($out/bin/ct completion fish) \
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:51]: > /nix/store/wq046wv3s3h4m2n9ak3j534di2pzra93-stdenv-linux/setup: line 114: /nix/store/asvv4qv70amf2bd4wfqi2z8j8qih6n3m-chart-releaser-1.6.1/bin/ct: No such file or directory
/nix/store/wq046wv3s3h4m2n9ak3j534di2pzra93-stdenv-linux/setup: line 114: /nix/store/asvv4qv70amf2bd4wfqi2z8j8qih6n3m-chart-releaser-1.6.1/bin/ct: No such file or directory
/nix/store/wq046wv3s3h4m2n9ak3j534di2pzra93-stdenv-linux/setup: line 114: /nix/store/asvv4qv70amf2bd4wfqi2z8j8qih6n3m-chart-releaser-1.6.1/bin/ct: No such file or directory


```suggestion
    installShellCompletion --cmd cr \
      --bash <($out/bin/cr completion bash) \
      --zsh <($out/bin/cr completion zsh) \
      --fish <($out/bin/cr completion fish) \
```

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      git
+      kubernetes-helm
+    ]}
+  '';
+
+  meta = with lib; {
+    description = "Hosting Helm Charts via GitHub Pages and Releases";
+    homepage = "https://github.com/helm/chart-releaser";
+    license = licenses.asl20;
+    # maintainers = with maintainers; [ atkinschang ];
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:64]: I think you should add yourself as a maintainer here.

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    owner = "helm";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-8+O9JErEB1Z/zlrWm975v5Qf0YG0lbPcjY5LlDKw8U4=";
+
+  };
+
+  # Don't run tests.
+  doCheck = false;
+  doInstallCheck = false;
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:27]:
```suggestion
```

Build succeeds without this line.

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    substituteInPlace pkg/config/config.go \
+      --replace "\"/etc/ct\"," "\"$out/etc/ct\","
+  '';
+
+  # https://github.com/helm/chart-releaser/blob/fa01315c4668d4fca627a5afc67409e31b27305c/.goreleaser.yml#L37
+  ldflags = [
+    "-w"
+    "-s"
+    "-X github.com/helm/chart-releaser/v3/ct/cmd.Version=${version}"
+    "-X github.com/helm/chart-releaser/v3/ct/cmd.GitCommit=${src.rev}"
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:41]: > nixpkgs on  feat/chart-releaser [!+] ✦ ➜  result/bin/cr version
GitVersion:  unreleased
GitCommit:   unknown
BuildDate:   unknown
GoVersion:   go1.22.2
Compiler:    gc
Platform:    linux/amd64
License:     Apache 2.0

Looks like it does not embed those string properly?

File: pkgs/by-name/ch/chart-releaser/package.nix [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    "-X github.com/helm/chart-releaser/v3/ct/cmd.BuildDate=19700101-00:00:00"
```
Comment (dawidd6) [inline pkgs/by-name/ch/chart-releaser/package.nix:42]:
```suggestion
    "-X github.com/helm/chart-releaser/cr/cmd.Version=${version}"
    "-X github.com/helm/chart-releaser/cr/cmd.GitCommit=${src.rev}"
    "-X github.com/helm/chart-releaser/cr/cmd.BuildDate=19700101-00:00:00"
```

File: pkgs/by-name/ch/chart-releaser/package.nix (hunk 2 above)
Comment (gnuion) [reply pkgs/by-name/ch/chart-releaser/package.nix:15]: The comment is not needed. Sorry! I should have deleted it.

**Comment by gnuion** [comment]
@dawidd6 thank you so much for your assistance. I hope to make error-free contributions in the future. I plan to spend more time learning nix and go in the future.

**Comment by dawidd6** [comment]
No prob, I'm no expert myself either :smile:. Glad I could help (and also learn something by the way).

**Pull request closed without merging**